packages/@aws-cdk/aws-events/lib/api-destination.ts

```diff
@@ -0,0 +1,130 @@
+import * as crypto from 'crypto';
+import { Duration, Lazy, Names, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { IConnection } from './connection';
+import { CfnApiDestination } from './events.generated';
+
+/**
+ * Supported HTTP operations.
+ */
+export enum HttpMethod {
+
+  /** POST */
+  POST = 'POST',
+  /** GET */
+  GET = 'GET',
+  /** HEAD */
+  HEAD = 'HEAD',
+  /** OPTIONS */
+  OPTIONS = 'OPTIONS',
+  /** PUT */
+  PUT = 'PUT',
+  /** PATCH */
+  PATCH = 'PATCH',
+  /** DELETE */
+  DELETE = 'DELETE',
+
+}
+
+/**
+ * The event Api Destination properties
+ */
+export interface ApiDestinationProps {
+  /**
+   * The ARN of the connection to use for the API destination
+   */
+  readonly connection: IConnection;
```

Comment:
```suggestion
  readonly connection?: IConnection;
```

The docs say that `connection` is optional. That probably corresponds to an anonymous request?

I wonder if we can make the API even easier by making it look like:

```ts
export interface ApiDestinationProps {
  /** 
   * The connection to use for this API Destination
   *
   * @default - Create a new connection from `authParameters` if those are given,
   * otherwise make an anonymous call.
   */  
  readonly connection?: IConnection;

  /**
   * Authentication to use for this API Destination (HTTP call)
   *
   * If `connection` is not supplied but `authParameters` are, implicitly
   * create a new `Connection` resource based on the specified auth parameters.
   *
   * This is convenient if you only have one or a few API destinations. If you want
   * to share authentication between multiple API destinations, create and pass
   * a `Connection` resource explicitly.
   *
   * @default - Use the existing connection supplied via `connection`, 
   * otherwise make an anonymous call.
   */
  readonly authParameters?: AuthParameters;
}
```

That way you don't even have to explicitly create a `Connection` object in the simple case.

Reply:
@rix0rrr I didn't find the documentation stating `connection` is optional. But in the AWS UI it's clearly required.
<img width="785" alt="Screen Shot 2021-08-04 at 4 18 06 PM" src="https://user-images.githubusercontent.com/958983/128249047-ddae7cd2-f376-4710-97be-91749b24afa9.png">